### ISSUE TITLE
NamedTuple field with default None is treated as type None

### ISSUE THREAD

**Issue by ironfoot** [microsoft/pyright#225]
**Describe the bug**
pyright thinks a NamedTuple field with default None is always type None.

**To Reproduce**
```python
from typing import NamedTuple, Optional


class Foo(NamedTuple):
    a: str
    b: Optional[str] = None

foo = Foo('one', 'two')
```

Output is
```bash
/Users/ironfoot/workspace/test.py
  10:18 - error: Argument of type 'Literal['two']' cannot be assigned to parameter 'b' of type 'None'
  Cannot assign to 'None'
```

**Expected behavior**
No error. Parameter `b` has type `str` not `None`.

**VS Code extension or command-line**
Both have same result.
Both on same version: pyright 1.0.57

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release of pyright.

**Comment by erictraut** [comment]
This bug is now fixed in 1.0.58, which I just published.
